apps/web/src/services/analytics/useMixPanelUserProperties.ts

```diff
@@ -0,0 +1,87 @@
+import { useMemo } from 'react'
+import { useCurrentChain } from '@/hooks/useChains'
+import useSafeInfo from '@/hooks/useSafeInfo'
+import { useAppSelector } from '@/store'
+import { selectTxHistory } from '@/store/txHistorySlice'
+import { isTransactionListItem } from '@/utils/transaction-guards'
+import { MixPanelUserProperty } from '@/services/analytics/mixpanel-events'
+
+export interface MixPanelUserProperties {
+  safe_address: string
+  safe_version: string
+  num_signers: number
+  threshold: number
+  networks: string[]
+  total_tx_count: number
+  last_tx_at: Date | null
+}
+
+export interface MixPanelUserPropertiesFormatted {
+  properties: Record<string, any>
+  networks: string[]
+}
+
+/**
+ * Hook to get formatted user properties for MixPanel tracking
+ *
+ * This hook collects Safe-related user properties that can be used for
+ * MixPanel user attribute tracking and cohort analysis.
+ * Returns both regular properties and networks separately for different MixPanel operations.
+ */
+export const useMixPanelUserProperties = (): MixPanelUserPropertiesFormatted | null => {
+  const { safe, safeLoaded } = useSafeInfo()
+  const currentChain = useCurrentChain()
+  const txHistory = useAppSelector(selectTxHistory)
+
+  return useMemo(() => {
+    if (!safeLoaded || !safe || !currentChain) {
+      return null
+    }
+
+    // Get current network name from chain
+    const currentNetworkName = currentChain.chainName.toLowerCase()
+
+    // Calculate transaction count and last transaction from history
+    let totalTxCount = 0
+    let lastTxAt: Date | null = null
+
+    if (txHistory.data?.results) {
+      const transactions = txHistory.data.results.filter(isTransactionListItem).map((item) => item.transaction)
+
+      totalTxCount = transactions.length
```

Comment:
This will not work because we only get the most recent 20 transactions in our request to the backend. You can instead use the `safe.nonce` which tells us how many transactions were executed on a safe.